xbmc/video/VideoDatabase.cpp

```diff
@@ -399,9 +399,17 @@ void CVideoDatabase::CreateViews()
 
   CLog::Log(LOGINFO, "create tvshow_view");
   std::string tvshowview = PrepareSQL("CREATE VIEW tvshow_view AS SELECT "
-                                     "  tvshow.*,"
-                                     "  path.idParentPath AS idParentPath,"
-                                     "  path.strPath AS strPath,"
+                                     "  tvshow.idShow AS idShow,"
+                                     "  tvshow.c00, tvshow.c01, tvshow.c02, tvshow.c03,"
+                                     "  tvshow.c04, tvshow.c05, tvshow.c06, tvshow.c07,"
+                                     "  tvshow.c08, tvshow.c09, tvshow.c10, tvshow.c11,"
+                                     "  tvshow.c12, tvshow.c13, tvshow.c14, tvshow.c15,"
+                                     "  tvshow.c16, tvshow.c17, tvshow.c18, tvshow.c19,"
+                                     "  tvshow.c20, tvshow.c21, tvshow.c22, tvshow.c23,"
+                                     "  tvshow.userrating AS userrating,"
+                                     "  tvshow.duration AS duration,"
+                                     "  min(path.idParentPath) AS idParentPath,"
+                                     "  min(path.strPath) AS strPath,"
```

Comment:
Is `min` necessary here? Shouldn't `strPath` be uniquely defined by the join?

Reply:
Hi, unfortunately not. Remember what we discussed in PR #15185 
table path may hold more than one rows for each tvshow, therefore we need to get whichever column is selected from table path into an aggregate value in order to simulate the arbitrary random row selected if we just used group by tvshow.idShow alone (with not a full group by clause that is). We need to make sure only one row is selected per tvshow so that we don't break current Kodi functionality.
I hope that makes sense.

Reply:
there's a different problem. I completely forgotten we expose also idparentpath.. that might be a problem since now we could have a path and a idparentpath from a different path.
I looked where idparentpath is used (i'm not 100% sure though) and with the tvshow_view is only used here:
https://github.com/xbmc/xbmc/blob/087521b803628221fda45626a8cca9809663c66b/xbmc/video/VideoDatabase.cpp#L10065
So we might actually drop that field from tvshow_view and fix the filter in that query..
Or we could simply change the tvshow_view to do the grouping in a subquery and remove it from the main level like this
```
SELECT  
  tvshow.idShow AS idShow, 
  tvshow.c00, tvshow.c01, tvshow.c02, tvshow.c03, 
  tvshow.c04, tvshow.c05, tvshow.c06, tvshow.c07, 
  tvshow.c08, tvshow.c09, tvshow.c10, tvshow.c11, 
  tvshow.c12, tvshow.c13, tvshow.c14, tvshow.c15, 
  tvshow.c16, tvshow.c17, tvshow.c18, tvshow.c19, 
  tvshow.c20, tvshow.c21, tvshow.c22, tvshow.c23, 
  tvshow.userrating AS userrating, 
  tvshow.duration AS duration, 
  path.idParentPath AS idParentPath, 
  path.strPath AS strPath, 
  tvshowcounts.dateAdded AS dateAdded, 
  lastPlayed, totalCount, watchedcount, totalSeasons,  
  rating.rating AS rating,  
  rating.votes AS votes,  
  rating.rating_type AS rating_type,  
  uniqueid.value AS uniqueid_value,  
  uniqueid.type AS uniqueid_type  
FROM tvshow 
  LEFT JOIN (SELECT idShow, MAX(idPath) as idPath FROM tvshowlinkpath
GROUP BY tvshowlinkpath.idShow) AS tvshowlinkpath ON 
    tvshowlinkpath.idShow=tvshow.idShow 
  LEFT JOIN path ON 
    path.idPath=tvshowlinkpath.idPath 
  INNER JOIN tvshowcounts ON 
    tvshow.idShow = tvshowcounts.idShow  
  LEFT JOIN rating ON 
    rating.rating_id=tvshow.c04  
  LEFT JOIN uniqueid ON 
    uniqueid.uniqueid_id=tvshow.c12 
```

Reply:
Yes, I thought that too, but if we said path information could be eventually ommited, I assumed that correctness of idparentpath should not be an issue...Anyway, however, what you suggest is even better, let me change it that way...

Reply:
Done, please kindly check. I also changed back to select tvshow.* and removed the GROUP BY clause.

Reply:
View with the latest build seems to be working fine...

![image](https://user-images.githubusercontent.com/23155784/50739016-a630ae00-11e2-11e9-9cdc-60db1b41797c.png)

Reply:
You need to bump version to 115 because someone might already have a 114 with yesterday build.. But how did you manage to create a new pr without conflicts starting from the old code? I'm not a git expert..

Reply:
Thanks. I changed the version to 115 now. To tell you the truth, via trial and error... :-) 
I restored my old branch which gave me the old latest code back. Then I changed the file, did git add . , then did git commit, and then I went to the web interface and compared the versions and from there I created a pull request...